packages/components/carousel/src/use-carousel.ts

```diff
@@ -481,17 +482,18 @@ export const useCarouselControl = ({
     }
   }, [carousel, isPrev])
 
-  const getControlProps: PropGetter<"button"> = useCallback(
-    (props = {}, ref = null) => ({
-      "aria-controls": id,
-      "aria-label": `Go to ${isPrev ? "previous" : "next"} slide`,
-      ...props,
-      ref,
-      disabled,
-      onClick: handlerAll(props.onClick, onClick),
-    }),
-    [disabled, id, onClick, isPrev],
-  )
+  const getControlProps: PropGetter<"button", { "aria-label": string }> =
+    useCallback(
+      (props = {}, ref = null) => ({
+        "aria-controls": id,
+        "aria-label": `Go to ${isPrev ? "previous" : "next"} slide`,
+        ...props,
+        ref,
+        disabled,
+        onClick: handlerAll(props.onClick, onClick),
+      }),
+      [disabled, id, onClick, isPrev],
+    )
```

Comment:
The type of `HTMLProps<"button">` is missing from the return type, is this a problem?
```ts
const getControlProps: PropGetter
(props?: HTMLProps<"button"> | undefined, ref?: React.Ref<any>) => {
    "aria-label": string;
} & React.RefAttributes<any>
```

Reply:
Thanks!!
You are right, my perception of `PropGetter` was wrong.

We have considered it, but cannot come up with a good idea as to how to fix it.
The following would seem to resolve the TS error and the problem you have identified.Do you have any other suggestions for a better fix?

```ts
  const getControlProps: PropGetter<"button", IconButtonProps> = useCallback(
    (props = {}, ref = null) => ({
      "aria-controls": id,
      "aria-label": `Go to ${isPrev ? "previous" : "next"} slide`,
      ...props,
      ref,
      disabled,
      onClick: handlerAll(props.onClick, onClick),
    }),
    [disabled, id, onClick, isPrev],
  )

// Result of type inference for `getControlProps
const getControlProps: PropGetter
(props?: HTMLProps<"button"> | undefined, ref?: React.Ref<any>) => IconButtonProps & React.RefAttributes<any>
```

Reply:
@odendayoko 
Good.